src/components/Table/Table.tsx

```diff
@@ -357,7 +357,8 @@ class Table extends React.Component<Props, State> {
 
   // Function to add checkbox in header as well
   addHeaderCheckbox = (): React.ReactElement<any> => {
-    return <TableHead componentStyle={{ width: 'auto' }}><Checkbox label="" checked={this.state.allRowChecked} onChange={this.toggleAllRowSelection} /></TableHead>;
+    const isAllChecked = ((this.state.totalRowCount - this.state.selectedRows.length) > 0 &&  this.state.totalRowCount > 0) ? true : false;
+    return <TableHead componentStyle={{ width: 'auto' }}><Checkbox label="" checked={isAllChecked && this.state.selectedRows.length > 0 ? true : (this.state.selectedRows.length > 0 && this.state.totalRowCount > 0 && this.state.totalRowCount === this.state.selectedRows.length) ? true : this.state.allRowChecked} indeterminante={isAllChecked} onChange={this.toggleAllRowSelection} /></TableHead>;
```

Comment:
@HiirenP Instead of defining the whole condition directly in jsx, its better o store the result in constant & then use it.